src/loadimg/utils.py

```diff
@@ -56,46 +79,55 @@ def download_image(url: str):
-        response.raise_for_status()  # Raise an exception for HTTP errors
+        response = requests.get(url, timeout=5)
+        response.raise_for_status()
         return Image.open(BytesIO(response.content))
     except requests.exceptions.RequestException as e:
         print(f"Error downloading image from {url}: {e}")
-        return None  # Return None if there's an error
-
-
-def load(img) -> Image.Image:
-    """loads the img and returns a pillow image"""
-    # base64 (str or bytes)
-    if isBase64(img):
-        image_bytes = base64.b64decode(img)
-        image_file = BytesIO(image_bytes)
-        return Image.open(image_file)
-    # file path or url
-    elif isinstance(img, str):
-        if os.path.isfile(img):
-            return Image.open(img)
-        else:
-            out = download_image(img)
-            if out is None:
-                raise ValueError(f"could not download {img}")
+        return None
+
+
+def load(img, input_type="auto") -> Image.Image:
+    """Loads an image from various sources and returns it as a Pillow Image."""
+    if input_type == "auto":
+        if isBase64(img):
+            input_type = "base64"
+        elif isinstance(img, str):
+            if os.path.isfile(img):
+                input_type = "file"
             else:
-                return out
-    # numpy array
-    elif isinstance(img, np.ndarray):
+                input_type = "url"
+        elif isinstance(img, np.ndarray):
+            input_type = "numpy"
+        elif isinstance(img, Image.Image):
+            input_type = "pil"
+        else:
+            raise ValueError("Could not determine input type automatically.")
+
+    if input_type == "base64":
+        if isinstance(img, str):
+            img = re.sub(r"^data:image\/[a-zA-Z]+;base64,", "", img)
+            image_bytes = base64.b64decode(img)
+            image_file = BytesIO(image_bytes)
+            return Image.open(image_file)
+        else:
+            image_bytes = base64.b64decode(img)
+            image_file = BytesIO(image_bytes)
+            return Image.open(image_file)
+    elif input_type == "file":
+        return Image.open(img)
+    elif input_type == "url":
+        out = download_image(img)
+        if out is None:
+            raise ValueError(f"could not download {img}")
+        else:
+            return out
+    elif input_type == "numpy":
         return Image.fromarray(img)
-    # pillow image
-    elif isinstance(img, Image.Image):
+    elif input_type == "pil":
         return img
     else:
-        available_types = [
-            i.__name__
-            for i in inspect.signature(load_img).parameters["img"].annotation.__args__
-        ]
-        available_types = ", ".join(available_types)
-        raise ValueError(
-            f"expected one of the following types :{available_types}, but got {type(img).__name__}"
-        )
+        raise ValueError(f"Invalid input type: {input_type}")
```

Comment:
same with this one, it would be better to inform the user what are the valid input types